From 69edc3dc2e8ca17d3789ff91320e4613e3e743cd Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Thu, 18 Jan 2018 22:55:45 +0800
Subject: [PATCH 2/4] power: Add known perf hint IDs

Author: Zhao Wei Liew <zhaoweiliew@gmail.com>
Date:   Tue Jun 21 10:44:49 2016 +0800

    power: msm8996: Support boost and perf profile hints

    Change-Id: I010d4d1d3fb495e523b065c11ca95c4305530f9f

Author: Zhao Wei Liew <zhaoweiliew@gmail.com>
Date:   Thu Oct 20 21:51:09 2016 +0800

    power: msm8937: Improve boost values

     * Bring boost values more in line with CAF's recommended values
     * Make performance profile values saner

    Change-Id: I9827d25a347a8ab959c766a71e30b3d29267983f

Author: Diogo Ferreira <diogo@underdev.org>
Date:   Sun May 7 11:00:12 2017 +0100

    power: Use the correct opcode for STOR_CLK_SCALE_DIS

    This is actually wrong, 0x42C0C000 is the opcode for SWAP_RATIO and
    0x42C10000 is the correct opcode for storage clock scale disabling.

    Change-Id: I6b1db525acf061ffe419011acd1f91525a27a35d

Author: dianlujitao <dianlujitao@lineageos.org>
Date:   Wed Sep 20 16:40:57 2017 +0800

    power: sdm660: Use declared enums

    Change-Id: I21691f9595f3730b9046012d1cae3951eb9da6a7

Author: dianlujitao <dianlujitao@lineageos.org>
Date:   Sat Sep 23 20:22:17 2017 +0800

    power: sdm660: Support boost and power profiles

     * perfd needs /vendor/etc/appboosts.xml on sdm660 platform.

    Change-Id: I92632da032ebd1a629fcc0b9c7e1811dee9117fe

Change-Id: I96e09d9840a5cb51383e5ca885f139599cc127ce
---
 performance.h | 95 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 95 insertions(+)

diff --git a/performance.h b/performance.h
index 5196076..23cc099 100644
--- a/performance.h
+++ b/performance.h
@@ -1,4 +1,5 @@
 /* Copyright (c) 2012, 2014, The Linux Foundation. All rights reserved.
+ * Copyright (C) 2018 The LineageOS Project
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are
@@ -46,6 +47,7 @@ enum SCREEN_DISPLAY_TYPE {
 };
 
 enum PWR_CLSP_TYPE {
+    ALL_CPUS_PWR_CLPS_DIS_V3 = 0x40400000, /* v3 resource */
     ALL_CPUS_PWR_CLPS_DIS = 0x101,
 };
 
@@ -103,6 +105,8 @@ enum CPU3_MAX_FREQ_LVL {
 };
 
 enum MIN_CPUS_ONLINE_LVL {
+    CPUS_ONLINE_MIN_BIG = 0x41000000, /* v3 resource */
+    CPUS_ONLINE_MIN_LITTLE = 0x41000100, /* v3 resource */
     CPUS_ONLINE_MIN_2 = 0x702,
     CPUS_ONLINE_MIN_3 = 0x703,
     CPUS_ONLINE_MIN_4 = 0x704,
@@ -111,6 +115,8 @@ enum MIN_CPUS_ONLINE_LVL {
 };
 
 enum MAX_CPUS_ONLINE_LVL {
+    CPUS_ONLINE_MAX_LIMIT_BIG = 0x41004000, /* v3 resource */
+    CPUS_ONLINE_MAX_LIMIT_LITTLE = 0x41004100, /* v3 resource */
     CPUS_ONLINE_MAX_LIMIT_1 = 0x8FE,
     CPUS_ONLINE_MAX_LIMIT_2 = 0x8FD,
     CPUS_ONLINE_MAX_LIMIT_3 = 0x8FC,
@@ -214,6 +220,7 @@ enum SCREEN_PWR_CLPS_LVL {
 
 enum THREAD_MIGRATION_LVL {
     THREAD_MIGRATION_SYNC_OFF = 0x1400,
+    THREAD_MIGRATION_SYNC_ON_V3 = 0x4241C000
 };
 
 enum INTERACTIVE_IO_BUSY_LVL {
@@ -222,6 +229,8 @@ enum INTERACTIVE_IO_BUSY_LVL {
 };
 
 enum SCHED_BOOST_LVL {
+    SCHED_BOOST_ON_V3 = 0x40C00000, /* v3 resource */
+    SCHED_BOOST_ON_V3_LVL = 0x40C02000, /* v3 resource */
     SCHED_BOOST_ON = 0x1E01,
 };
 
@@ -261,6 +270,92 @@ enum CPU7_MAX_FREQ_LVL {
     CPU7_MAX_FREQ_NONTURBO_MAX = 0x260A,
 };
 
+enum SCHED_PREFER_IDLE {
+    SCHED_PREFER_IDLE_DIS_V3 = 0x40C04000,
+    SCHED_PREFER_IDLE_DIS = 0x3E01,
+};
+
+enum SCHED_MIGRATE_COST_CHNG {
+    SCHED_MIGRATE_COST_SET = 0x3F01,
+};
+
+/**
+ * MPCTL v3 opcodes
+ */
+enum MAX_FREQ_CLUSTER_BIG {
+    MAX_FREQ_BIG_CORE_0         = 0x40804000,
+    MAX_FREQ_BIG_CORE_0_LVL   = 0x40806000,
+};
+
+enum MAX_FREQ_CLUSTER_LITTLE {
+    MAX_FREQ_LITTLE_CORE_0      = 0x40804100,
+    MAX_FREQ_LITTLE_CORE_0_LVL  = 0x40806100,
+};
+
+enum MIN_FREQ_CLUSTER_BIG {
+    MIN_FREQ_BIG_CORE_0         = 0x40800000,
+    MIN_FREQ_BIG_CORE_0_LVL     = 0x40802000,
+};
+
+enum MIN_FREQ_CLUSTER_LITTLE {
+    MIN_FREQ_LITTLE_CORE_0      = 0x40800100,
+    MIN_FREQ_LITTLE_CORE_0_LVL  = 0x40802100,
+};
+
+enum INTERACTIVE_CLUSTER_BIG {
+    ABOVE_HISPEED_DELAY_BIG     = 0x41400000,
+    ABOVE_HISPEED_DELAY_BIG_LVL = 0x41402000,
+    GO_HISPEED_LOAD_BIG         = 0x41410000,
+    HISPEED_FREQ_BIG            = 0x41414000,
+    IO_IS_BUSY_BIG              = 0x4141A000,
+    TARGET_LOADS_BIG            = 0x41420000,
+    IGNORE_HISPEED_NOTIF_BIG    = 0x41438000,
+};
+
+enum INTERACTIVE_CLUSTER_LITTLE {
+    ABOVE_HISPEED_DELAY_LITTLE  = 0x41400100,
+    GO_HISPEED_LOAD_LITTLE      = 0x41410100,
+    HISPEED_FREQ_LITTLE         = 0x41414100,
+    TARGET_LOADS_LITTLE         = 0x41420100,
+    IGNORE_HISPEED_NOTIF_LITTLE = 0x41438100,
+};
+
+enum CPUBW_HWMON {
+    CPUBW_HWMON_MIN_FREQ        = 0x41800000,
+    CPUBW_HWMON_MIN_FREQ_LVL    = 0x41802000,
+    CPUBW_HWMON_V1              = 0x4180C000,
+    LOW_POWER_CEIL_MBPS         = 0x41810000,
+    LOW_POWER_IO_PERCENT        = 0x41814000,
+    CPUBW_HWMON_SAMPLE_MS       = 0x41820000,
+};
+
+enum SCHEDULER {
+    SCHED_SMALL_TASK_DIS        = 0x40C0C000,
+    SCHED_IDLE_LOAD_DIS         = 0x40C10000,
+    SCHED_IDLE_NR_RUN_DIS       = 0x40C14000,
+    SCHED_IDLE_NR_RUN           = 0x40C2C000,
+    SCHED_IDLE_NR_RUN_LVL       = 0x40C2E000,
+    SCHED_IDLE_RESTRICT_CLUSTER = 0x40C34000,
+    SCHED_IDLE_RESTRICT_CLUSTER_LVL = 0x40C36000,
+    SCHED_GROUP_ON              = 0x40C28000,
+    SCHED_GROUP_UP_MIGRATE      = 0x40C54000,
+    SCHED_GROUP_DOWN_MIGRATE    = 0x40C58000,
+    SCHED_FREQ_AGGR_THRH_LVL    = 0x40C5E000,
+};
+
+enum STORAGE {
+    STOR_CLK_SCALE_DIS          = 0x42C10000,
+    STOR_CLK_SCALE_DIS_LVL      = 0x42C12000,
+};
+
+enum GPU {
+    GPU_MIN_PWRLVL_BOOST        = 0x42804000,
+    GPU_MIN_FREQ                = 0x4280C000,
+    GPU_MIN_FREQ_LVL            = 0x4280E000,
+    GPU_MAX_FREQ                = 0x42810000,
+    GPUBW_MIN_FREQ              = 0x42814000,
+};
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.7.4

